observability/correlation-id.md

```diff
@@ -0,0 +1,41 @@
+# Correlation IDs
+
+## The Need
+
+In a distributed system architecture (microservice architecture), it is highly difficult to understand a single end-to-end customer transaction flow through the various components. 
+
+Here are some the general challenges -
+* It becomes challenging to understand the end-to-end behavior of a client request entering the application.
+*  Aggregation: Consolidating logs from multiple components and making sense out of these logs is difficult, if not impossible.
+*  Cyclic dependencies on services, course of events and asynchronous requests are not easily deciphered.
+* While troubleshooting a request, the diagnostic context of the logs are very important to get to the root of the problem.
+
+## Solution
+
+A Correlation ID is a unique identifier that is added to the very first interaction (incoming request) to  identify the context and is passed to all components that are involved in the transaction flow. Correlation ID becomes the glue that binds the transaction together and helps to draw a overall picture of events.
+
+*Note: Before implementing your own Correlation ID, investigate if your telemetry tool of choice provides a auto-generated Correlation ID and that it serves the purposes of your application. For instance, [Application Insights](https://docs.microsoft.com/en-us/azure/azure-monitor/app/auto-collect-dependencies) offers dependency auto-collection for some application frameworks*  
```

Comment:
Nit: This would look a little better as a quote than italics .. i.e. by removing the *..* and instead starting with a >